assembly/src/parsers/context.rs

```diff
@@ -25,6 +25,13 @@ pub struct ParserContext {
     pub local_constants: LocalConstMap,
 }
 
+/// Call target context that holds the information about the target of a call. This information
+/// is subsequently used to resolve the call target.
+pub enum CallTargetContext<'a> {
+    MastRoot(RpoDigest),
+    ProcedureId((&'a str, Option<&'a str>)),
+}
```

Comment:
It seems like `ProcedureId` variant now covers 2 variants: `name` (for local procs) and `name + module` (for imported procs). Maybe we should separate these 3 variants even at this level like so:
```Rust
pub enum CallTarget<'a> {
    MastRoot(RpoDigest),
    ProcedureName(&'a str),
    ProcedurePath { name: &'a str, module: &'a str },
}
```
We could even rename this to `InvocationTarget` and use it with `exec` and `syscall` instructions as well. In this case, we could implement `parse` function like:

```Rust
impl<'a> InvocationTarget<'a> {
    pub fn parse(label: &'a str, token: &'a Token) -> Result<Self, ParsingError> {
        todo!()
    }
}
```
And this could replace `validate_proc_invocation_label()` function.

Reply:
Good call!  I've updated the implementation accordingly.